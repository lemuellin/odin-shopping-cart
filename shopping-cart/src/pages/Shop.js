import ShopItem from "../component/ShopItem";

const Shop = () => {
    return (
      <div>
        Shop Page
        <ShopItem/>
      </div>
    );
  }
  
  export default Shop;