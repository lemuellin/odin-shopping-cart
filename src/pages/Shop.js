import React from 'react';
import ShopItem from '../component/ShopItem';
import '../style/shop.css';

import Echeveria from '../asset/Echeveria';
import Cactus from '../asset/Cactus';
import Aloe from '../asset/Aloe';

import Cart from '../component/Cart';
import cartIcon from '../asset/icons/bag.png';

const Shop = (props) => {
    let cartStatus;
    
    const showCart = () => {
        cartStatus = (cartStatus === 'on') ? 'off' : 'on';
        let displayCart = (cartStatus === 'on') ? 'flex' : 'none';
        let displayItem = (cartStatus === 'on') ? 'none' : 'grid';
        document.getElementsByClassName('cartList')[0].style.display = displayCart;
        document.getElementsByClassName('allShopItems')[0].style.display = displayItem;
    };

    return(
        <div className='shop'>
            <div className='status' onClick={showCart}>
                <div className='itemCount'>{props.itemCount}</div>
                <img src={cartIcon} alt='cart icon' className='cartIcon'/>
            </div>
            <Cart content={props.cart} inputField={props.handleChange} minusButton={props.itemDecrease} plusButton={props.itemIncrease} removeButton={props.itemRemove} showCart={showCart}/>
            <div className='allShopItems'>
                <div className='shopSection'>
                    <h2 className='itemsTitle'>ECHEVERIA</h2>
                    <p>Echeveria is a family of rose-shaped succulents native to the semi-desert regions of Central America. Because of its charming leaves' structure, Echeverias are one of the most well-known succulents. They are super easy to grow and take care of.</p>
                    <div className='itemsGrid'>
                        <ShopItem Succulent={Echeveria} addToCart={props.addToCart}/>
                    </div>
                </div>

                <div className='shopSection'>
                    <h2 className='itemsTitle'>CACTUS</h2>
                    <p>A collection of beautiful cacti that have scales and spines instead of leaves to adapt to extremely dry desert areas. Most of them have thick fleshy stems or branches to conserve water, so they can tolerate drought very well. A cactus is a great addition to the home thanks to its striking shape, size, and flowers.</p>
                    <div className='itemsGrid'>
                        <ShopItem Succulent={Cactus} addToCart={props.addToCart}/>
                    </div>
                </div>
                
                <div className='shopSection'>
                    <h2 className='itemsTitle'>ALOE</h2>
                    <p>Aloe leaves are thick and fleshy, triangle-shaped. The inside of Aloe leaves is full of slimy, water-filled tissue. These pretty succulents are easy to care for, grow quickly, can tolerate drought, and perfect for indoor or outdoor gardens.</p>
                    <div className='itemsGrid'>
                        <ShopItem Succulent={Aloe} addToCart={props.addToCart}/>
                    </div>
                </div>

            </div>
        </div>
    )
}

export default Shop;