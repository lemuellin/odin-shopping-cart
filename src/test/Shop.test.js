import React from 'react';
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from '@testing-library/user-event';
import Shop from '../pages/Shop';

describe("Shop Component", () => {
    it("render correct items", () => {
        const { container } = render(<Shop/>);
        expect(screen.getByText("ECHEVERIA")).toBeInTheDocument();
        expect(screen.getByText("A rare low-growing succulent that forms shrubs of thick chubby leaves.")).toBeInTheDocument();
        expect(screen.getByText("CACTUS")).toBeInTheDocument();
        expect(screen.getByText("GOLDEN BALL CACTUS")).toBeInTheDocument();
        expect(screen.getByText("ALOE")).toBeInTheDocument();
        expect(screen.getByText("Is distinguished by its thick, pointed, and juicy green leaves with a toothed border.")).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});