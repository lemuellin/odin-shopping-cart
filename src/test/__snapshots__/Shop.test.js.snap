// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Component render correct items 1`] = `
<div>
  <div>
    <div
      class="status"
    >
      <div
        class="itemCount"
      >
        0
      </div>
      <img
        alt="cart icon"
        class="cartIcon"
        src="bag.png"
      />
    </div>
    <div
      class="cartList"
    >
      <div
        class="cartCard"
      >
        <h2
          class="itemsTitle"
        >
          YOUR CART
        </h2>
        <div />
        <div
          class="total"
        >
          <h4>
            Subtotal: 
          </h4>
          <h4>
            $ 0.00
          </h4>
          <h4>
            Shipping:
          </h4>
          <h4>
            free
          </h4>
          <h4>
            Tax:
          </h4>
          <h4>
            $ 0.00
          </h4>
          <h3>
            Grand Total: 
          </h3>
          <h3
            data-testid="gTotal"
          >
            $ 0.00
          </h3>
        </div>
        <div
          class="buttons"
        >
          <button
            class="backToShop"
          >
            Back To Shop
          </button>
          <button
            class="checkout"
          >
            Checkout
          </button>
        </div>
      </div>
    </div>
    <div
      class="allShopItems"
    >
      <div
        class="shopSection"
      >
        <h2
          class="itemsTitle"
        >
          ECHEVERIA
        </h2>
        <p>
          Echeveria is a family of rose-shaped succulents native to the semi-desert regions of Central America. Because of its charming leaves' structure, Echeverias are one of the most well-known succulents. They are super easy to grow and take care of.
        </p>
        <div
          class="itemsGrid1"
        >
          <div
            class="itemCard"
            id="1"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="1.jpg"
              />
              <div
                class="shopItemDescription"
              >
                It has beautiful pale gray green rosettes and powdery pointed symmetrical leaves.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                ECHEVERIA LOLA
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="2"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="2.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Its leaves are fleshy, spike-shaped and covered with a powdery green white layer.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                WOOLLY SENECIO
              </h3>
              <h4>
                $ 8
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="3"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="3.jpg"
              />
              <div
                class="shopItemDescription"
              >
                A rare low-growing succulent that forms shrubs of thick chubby leaves.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                CRASSULA HERREI
              </h3>
              <h4>
                $ 6
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="4"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="4.jpg"
              />
              <div
                class="shopItemDescription"
              >
                The white felt covering the surface of its upright cylindrical leaves.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                SENECIO COCOON PLANT
              </h3>
              <h4>
                $ 6
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="5"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="5.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Has plump, blue-green rounded leaves ending in a point and growing in a tight clump.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                ECHEVERIA ELEGANS BLUE
              </h3>
              <h4>
                $ 6
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shopSection"
      >
        <h2
          class="itemsTitle"
        >
          CACTUS
        </h2>
        <p>
          A collection of beautiful cacti that have scales and spines instead of leaves to adapt to extremely dry desert areas. Most of them have thick fleshy stems or branches to conserve water, so they can tolerate drought very well. A cactus is a great addition to the home thanks to its striking shape, size, and flowers.
        </p>
        <div
          class="itemsGrid2"
        >
          <div
            class="itemCard"
            id="6"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="6.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Primarily suited for growing in garden rockery settings, desert type landscapes, patios and botanical gardens.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                GOLDEN BARREL CACTUS
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="7"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="7.jpg"
              />
              <div
                class="shopItemDescription"
              >
                It is distinctive for its long, trailing stems, fine spines that grow to 6 feet in length at maturity.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                MINI RAT'S TAIL CACTUS
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="8"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="8.jpg"
              />
              <div
                class="shopItemDescription"
              >
                A rare cactus that has no spikes but white hairy scales that give it a characteristic chalk-white or silvery-grey appearance.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                BISHOP'S CAP CACTUS
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="9"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="9.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Also called Golden Ball, Lemon Ball, or Yellow Tower. It's covered with harmless thin yellow spines.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                GOLDEN BALL CACTUS
              </h3>
              <h4>
                $ 7
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="10"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="10.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Frequently produce flowers in beautiful shades of yellow, red, orange, or pink.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                Ball Cactus
              </h3>
              <h4>
                $ 6
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="11"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="11.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Is indigenous to Mexico, a bushy succulent shrub with light green, egg-shaped, fleshy pads.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                HAIRY ROGER CACTUS
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="12"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="12.jpg"
              />
              <div
                class="shopItemDescription"
              >
                A slender, erect, arching or somewhat pendent, bushy, cactus, up to 2 feet (60 cm) tall.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                DANCING BONES CACTUS
              </h3>
              <h4>
                $ 8
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="13"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="13.jpg"
              />
              <div
                class="shopItemDescription"
              >
                The pads are covered with very small brownish-cinnamon colored spines or glochids. Very compact and self branching.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                CINNAMON CACTUS
              </h3>
              <h4>
                $ 12
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="14"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="14.jpg"
              />
              <div
                class="shopItemDescription"
              >
                These cacti are not toxic. It can be placed near pets and children without any worries.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                MOON CACTUS
              </h3>
              <h4>
                $ 12
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="15"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="15.jpg"
              />
              <div
                class="shopItemDescription"
              >
                It can grow quite large, spreading out over a thick columnar trunk, stems cresting and creating waves as it grows.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                CRESTED BLUE CANDLE
              </h3>
              <h4>
                $ 10
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="16"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="16.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Composed of a tall, thick, fluted, columnar stem, often with several large branches curving upward.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                SAGUARO CACTUS
              </h3>
              <h4>
                $ 12
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="shopSection"
      >
        <h2
          class="itemsTitle"
        >
          ALOE
        </h2>
        <p>
          Aloe leaves are thick and fleshy, triangle-shaped. The inside of Aloe leaves is full of slimy, water-filled tissue. These pretty succulents are easy to care for, grow quickly, can tolerate drought, and perfect for indoor or outdoor gardens.
        </p>
        <div
          class="itemsGrid3"
        >
          <div
            class="itemCard"
            id="17"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="17.jpg"
              />
              <div
                class="shopItemDescription"
              >
                A cute aloe species with pink ridges. The textured leaves are dark green mixed with soft green markings.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                PINK BLUSH ALOE
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="18"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="18.jpg"
              />
              <div
                class="shopItemDescription"
              >
                It spreads out in clusters along the ground as it grows. Its spear-shaped leaves have harmless off-white to yellow “teeth” along the ridges.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                ALOE PERFOLIATA
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="19"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="19.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Also called Devil’s Ivy, it has shiny, heart-shaped leaves and comes in a number of natural and cultivated varieties.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                POTHOS
              </h3>
              <h4>
                $ 5
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
          <div
            class="itemCard"
            id="20"
          >
            <div
              class="imgContainer"
            >
              <img
                alt="Succulent Photos"
                class="shopItemPhoto"
                src="20.jpg"
              />
              <div
                class="shopItemDescription"
              >
                Is distinguished by its thick, pointed, and juicy green leaves with a toothed border.
              </div>
            </div>
            <div
              class="item"
            >
              <h3>
                ALOE VERA
              </h3>
              <h4>
                $ 12
              </h4>
              <button
                class="addToCart"
                type="button"
              >
                Add To Cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
